### PR TITLE
AST-2400 - fix: Related post option showing notice on user site

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

- After enabling the related post on a single post, some error is showing on frontend

Warning: Trying to access array offset on the value of type int in /data/web/virtuals/303317/virtual/www/domains/topromanticgetaways.com/wp-includes/class-wp-list-util.php on line 170

Warning: Trying to access array offset on the value of type int in /data/web/virtuals/249059/virtual/www/domains/funtripswithkids.com/wp-includes/class-wp-list-util.php on line 170

Steps To Reproduce: 

1. WordPress latest version install 
2. and on the related post on a single post 
3. On the frontend, it's showing a notice issue.

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by nathanrodrigues2111 — APPROVED** [review]

**Review by nathanrodrigues2111 — COMMENTED** [review]
Please 
- add changelogs
- check Code Analysis / analysis
- check PR Title Checker / check 
- update branch for next release @yssambare12

**Comment by yssambare12** [comment]
> Please
> 
> * add changelogs
> * check Code Analysis / analysis
> * check PR Title Checker / check
> * update branch for next release @yssambare12

@nathanrodrigues2111 Changes are done.

**Comment by neerajmasai** [comment]
Please resolve the conflicts

**Comment by yssambare12** [comment]
> Please resolve the conflicts

Done

File: changelog.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ v4.0.0 (Unreleased)
 - Fix: Off-canvas - Full screen - Mobile Menu gets very long scroll when Elementor plugin is activated.
 - Fix: Full Width/Contained layout Layout is showing extra padding above in the Transparent Header.
 - Fix: Block editor - Getting "wp.data.select( 'core/editor' )getBlocks()" console warning in editor as being deprecated in WordPress version 5.3.
+- Fix: Related post option showing notice on user site.
```
Comment (imnavanath) [inline changelog.txt:11]: @yssambare12 this seems to be not clear, can you little bit more elaborate so it can be understandable?

Comment (yssambare12) [reply changelog.txt]: @imnavanath Updated.

File: changelog.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ v4.0.0 (Unreleased)
 - Fix: Full Width/Contained layout Layout is showing extra padding above in the Transparent Header.
 - Fix: Block editor - Getting "wp.data.select( 'core/editor' )getBlocks()" console warning in editor as being deprecated in WordPress version 5.3.
 - Fix: Block editor - Button background colour gets overridden by core styles for Legacy block editor setup.
+- Fix: Customizer - After enabling related post option from customizer it's showing errors on post pages.
```
Comment (imnavanath) [inline changelog.txt:12]:
```suggestion
- Fix: Related Posts showing "Trying to access array offset on value of type int" error on Latest Posts/Blog pages.
```

**Review by imnavanath — APPROVED** [review]

File: inc/modules/related-posts/class-astra-related-posts-markup.php [hunk 3]
```diff
@@ -382,8 +383,9 @@ public function astra_get_related_posts_by_query( $post_id ) {
 		}
 
 		$query_args = apply_filters( 'astra_related_posts_query_args', $query_args );
-
-		return new WP_Query( $query_args );
+		/** @psalm-suppress PossiblyInvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+		return get_posts( $query_args );
```
Comment (imnavanath) [inline inc/modules/related-posts/class-astra-related-posts-markup.php:387]: @yssambare12 can we avoid `get_posts` & use wp_query only. As get_posts overrides passed defaults & alters query cache  as well.

Comment (yssambare12) [reply inc/modules/related-posts/class-astra-related-posts-markup.php:387]: As the wp-query global is an object, the ID we need is not getting. That's why I use get_post here.

**Comment by yssambare12** [comment]
We fixed this issue for wordpress 6.1 updated, But eventually, after Wordpress updated to 6.1.1, this issue was resolved. So closing this PR.

**Pull request closed without merging**